File: .agents/skills/create-capabilities-python/templates/capabilities.py
from shared.<domain>.taxonomy_<name>_vo import <VO>
from shared.<domain>.contract_<name>_protocol import I<Name>Protocol


# ─── Block 1: Class Definition & Constructor ──────────────
class Capabilities<Name>(I<Name>Protocol):
    def __init__(self, /* DI params */) -> None:
        # DI fields use protocol interfaces
        # Value fields use shared VOs
        ...

    # ─── Block 2: Public Contract (domain protocol ONLY) ──
    def method_name(self, param: <VO>) -> None:
        # domain behavior
        ...

    # ─── Block 3: Dunder Methods, Factories & Helpers ─────
    def __repr__(self) -> str:
        return "Capabilities<Name>()"

    @classmethod
    def create_default(cls) -> "Capabilities<Name>":
        return cls()

File: .agents/skills/create-capabilities-rust/templates/capabilities.rs
use std::sync::Arc;

use shared::<domain>::taxonomy_<name>_vo::<VO>;
use shared::<domain>::contract_<name>_protocol::I<Name>Protocol;

// ─── Block 1: Struct Definition ───────────────────────────
pub struct Capabilities<Name> {
    // DI fields use Arc<dyn Trait>
    // Value fields use shared VOs
}

// ─── Block 2: Public Contract (domain protocol ONLY) ──────
impl I<Name>Protocol for Capabilities<Name> {
    // public contract methods only
}

// ─── Block 3: Constructors, Std Traits & Helpers ─────────
impl Default for Capabilities<Name> {
    fn default() -> Self {
        Self
    }
}

impl Capabilities<Name> {
    pub fn new(/* DI params */) -> Self {
        Self {
            // ...
        }
    }

    // private helpers here
}
